app/controllers/milestones_controller.rb

```diff
@@ -0,0 +1,62 @@
+class MilestonesController < ApplicationController
+  respond_to :html, :js
+
+  def index
+    @milestones =Milestone.joins(:events).where(goal_id: params[:id])
```

Comment:
It's weird to have params[:id] in your index like this. In fact I'm not sure this would work. The two routes that map to this are 
goal_milestones GET    /goals/:goal_id/milestones(.:format)          milestones#index  
and  
milestones GET    /milestones(.:format)                         milestones#index  

In the first you'd have params[:goal_id] and in the second there's no param